src/Sarif/Writers/SarifLogger.cs

```diff
@@ -149,11 +149,9 @@ private static string Redact(string text, IEnumerable<string> tokensToRedact)
 
         private void SetSarifLoggerVersion(Tool tool)
         {
-            tool.Properties = tool.Properties ?? new Dictionary<string, string>();
```

Comment:
Example of a place where we don't need the initialization because `SetProperty` takes care of it.